python/graphstorm/gconstruct/utils.py

```diff
@@ -329,8 +547,8 @@ def _merge_arrs(arrs, tensor_path):
         for arr in arrs:
             out_arr[row_idx:(row_idx + arr.shape[0])] = arr[:]
             row_idx += arr.shape[0]
-        return out_arr
-    elif isinstance(arrs[0], HDF5Array):
+        return ExtNumpyWrapper(tensor_path, out_arr.shape, out_arr.dtype)
+    elif isinstance(arrs[0], ExtMemArrayWrapper):
         arrs = [arr.to_numpy() for arr in arrs]
         return np.concatenate(arrs)
```

Comment:
Does np.concatenate support concatenating memmaped arrays?

Reply:
we have called to_numpy. it should return normal numpy arrays.

Reply:
I c